dbms/src/Storages/MergeTree/MergeTreeReader.cpp

```diff
@@ -447,15 +467,14 @@ void MergeTreeReader::readData(const String & name, const IDataType & type, ICol
 					name,
```

Comment:
Here is the problem: if required_internal_size is zero, then this readData is not called and cur_mark_idx is not updated for the stream corresponding to the array column. This leads to unnecessary seeks.